app/ui/src/HttpArchiveTest.elm

```diff
@@ -0,0 +1,327 @@
+
+
+log =
+    { entries = [ entry, entry ]
+    }
+
+
+all : Test
+all =
+    describe "HttpArchive"
+        [ describe "Parsing a Request"
+            [ test "decodes a request with no headers" <|
+                \_ ->
+                    let
+                        input =
+                            requestJson
+
+                        output =
+                            (parse HttpArchive.request input)
+
+                        expected =
+                            Ok request
+                    in
+                        Expect.equal expected output
+            , test "decodes a request with headers" <|
+                \_ ->
+                    let
+                        input =
+                            requestWithHeadersJson
+
+                        output =
+                            (parse HttpArchive.request input)
+
+                        expected =
+                            Ok requestWithHeaders
+                    in
+                        Expect.equal expected output
+            ]
+        , describe "Parsing a Response"
+            [ test "decodes a response with no headers" <|
+                \_ ->
+                    let
+                        input =
+                            responseJson
+
+                        output =
+                            (parse HttpArchive.response input)
+
+                        expected =
+                            Ok response
+                    in
+                        Expect.equal expected output
+            , test "decodes a response with headers" <|
+                \_ ->
+                    let
+                        input =
+                            responseWithHeadersJson
+
+                        output =
+                            (parse HttpArchive.response input)
+
+                        expected =
+                            Ok responseWithHeaders
+                    in
+                        Expect.equal expected output
+            ]
+        , describe "Parsing an Entry"
+            [ test "decodes an entry with timings, request and response" <|
+                \_ ->
+                    let
+                        input =
+                            entryJson
+
+                        output =
+                            (parse HttpArchive.entry input)
+
+                        expected =
+                            Ok entry
+                    in
+                        Expect.equal expected output
+            ]
+        , describe "Parsing a Log"
+            [ test "decodes a log with two full entries" <|
+                \_ ->
+                    let
+                        input =
+                            logJson
+
+                        output =
+                            (parse HttpArchive.log input)
+
+                        expected =
+                            Ok log
+                    in
+                        Expect.equal expected output
+            ]
+        ]
```

Comment:
Who's convention? I just find them a little more readable. But then abstracting that I find less readable again, because I have to first think about what the mapping means. 

I want tests to be as obvious and explicit as they possibly can be. Preferably no cleverness whatsoever.